Library/Homebrew/software_spec.rb

```diff
@@ -53,7 +53,7 @@ def owner=(owner)
     @resource.owner = self
     resources.each_value do |r|
       r.owner     = self
-      r.version ||= version
+      r.version ||= version.dup
```

Comment:
My main thought is we should discard the commit of HEAD-commit from formula when passing to resources  

We could certainly add `dup` to non-stable version as `... : version.dup`.